sw/host/cryptotest/testvectors/parsers/nist_cavp_aes_kw_parser.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Parser for converting NIST AES-KWP testvectors to JSON.
+
+"""
+
+import argparse
+import sys
+import json
+import jsonschema
+
+from cryptotest_util import parse_rsp, str_to_byte_array
+
+
+def parse_testcases(args) -> None:
+    raw_testcases = parse_rsp(args.src)
+    testcases = list()
+    test_case_id = 1
+    for test_key in raw_testcases.keys():
+        for test_vec in raw_testcases[test_key]:
+            tmode = "null"
+            if args.transformation_mode == "inverse":
+                tmode = "forward" if args.operation == "decrypt" else "inverse"
+            testcase = {
+                "test_case_id": test_case_id,
+                "operation": args.operation.lower(),
+                "transformation_mode": tmode,
+                "padding": args.padding,
+                "key_len": args.key_len,
+                "key": str_to_byte_array(test_vec["K"]),
+                "ciphertext": str_to_byte_array(test_vec["C"]),
+                "plaintext": str_to_byte_array(test_vec["P"]) if "P" in test_vec else [],
```

Comment:
Just to let you know, when you include an optional parameter like this, you'll need to add `#[serde(deafult)]` to the variable in the test case type in the Rust test harness once you get there, otherwise the deserialization will fail for test cases that lack this parameter.

Reply:
Is the code selection part for your comment correct? I can't see any optional parameter in this code selection. Or you meant this:

```python
    parser.add_argument(
        "--padding",
        type = bool,
        default = False,
        help = "Existence of padding."
    )
    ```

Reply:
Never mind, I see you're passing an array as the default. I was referring to if you did not include `plaintext` at all in the JSON test vector when it does not exist in the source test vector.